### PR TITLE
Adopt logging best practices

### PR DESCRIPTION
**Pull request by heitorlessa**
*Issue #, if available:* #16 #21 

## BREAKING CHANGE

`logger_setup` and `logger_inject_lambda_context` had to be deprecated in favour of `Log` class - The user experience remains the same so it's easier swap. If you try to use any of these functions it'll raise an `DeprecationWarning` exception.

While we're on Beta we strive to make backwards compatible changes, however this wasn't the case - The former implementation impact all loggers (root) and can't be easily updated to allow both decorators to work cleanly, and without side effects.

**UX**

UX stays the same except we now use a class `Logger` instead of functions.

```python
from aws_lambda_powertools.logging import Logger

logger = Logger(service="payment", level="INFO") 
# logger = Logger() # implicit config via env vars 

@logger.inject_lambda_context
def handler(event, context)
  logger.info("Collecting payment")
  logger.info({ 
    "operation": "collect_payment",
    "charge_id": event['charge_id'] 
  })
```

**NEW - Appending new keys to structured log**

```python
...
@logger.inject_lambda_context
def handler(event, context):
    if "payment_id" in event:
        logger.structure_logs(append=True, payment_id=event["payment_id"])
    logger.info("Collecting payment")
```

*Description of changes:*

* [x] Use a Null Handler to cut noise in customers' logging
* [x] Create top-level logger `aws-lambda-powertools`
* [x] Refactor Lambda Logging module to prevent root logging changes
* [x] ~~Prevent unnecessary exceptions from Lambda Logging module (e.g. KeyError)~~ 
* [x] Create Logger class and reuse same UX as before
* [x] Raise deprecated exception for standalone `logger_setup`, and `logger_inject_lambda_context`
* [x] Update docstring
* [x] Allow appending log keys to structured log
* [x] Update docs to reflect the change
* [x] Document package logger and how to enable debug
* [x] Update tests
* [x] Talk about deprecation and root logger

**Optionally**

* ~~[x] Create a `POWERTOOLS_LOG_LEVEL` environment variable~~
* ~~[x] Set package log level if POWERTOOLS_LOG_LEVEL is set~~
* ~~[ ] Set boto logs via env var~~
    - Best handled by customer's code
* ~~[ ] Accept an existing logger in logger setup~~
    - KISS
* [x] Complete coverage

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by heitorlessa** [comment]
@marcioemiranda - Let me know if I missed anything else as I'm working on logger this week to fix logging level along with other suboptimal practices

**Comment by heitorlessa** [comment]
@jfuss When you can, may I have your review? This adopts the logging best practices as we discussed

File: python/aws_lambda_powertools/logging/logger.py [hunk 1]
```diff
@@ -278,3 +249,226 @@ def __build_dimensions(**dimensions) -> str:
...
+        across future log statements.
+
+        Last keyword argument and value wins if duplicated.
+
+        Parameters
+        ----------
+        append : bool, optional
+            [description], by default False
+        """
+        self.handler.setFormatter(JsonFormatter(**self._default_log_keys, **kwargs))
```
[the remark this answers is not in the document]
Comment (jfuss) [reply python/aws_lambda_powertools/logging/logger.py:460]: What do additional keys have to do with how the Json data is formatted? I am thinking of the case where Company X has a different way they want the data to be formatted, which doesn't have anything to do with what the data actually is in the Json. Or maybe they don't want Json at all and prefer XML (for whatever reason). I don't think this blocks anything here but something to maybe think about in the future.

File: python/README.md [hunk 2]
```diff
@@ -113,6 +114,8 @@ tracer = Tracer(auto_patch=False) # new instance using existing configuration wi
 
 ### Logging
 
+> **NOTE** `logger_setup` and `logger_inject_lambda_context` are deprecated and will be completely removed once it's GA.
```
Comment (jfuss) [inline python/README.md:117]: Aren't they kind of already removed since those methods just raise an exception?

**Pull request merged**